Assets/00_Content/Scripts/Interactables/Table.cs

```diff
@@ -138,5 +139,21 @@ public class Table : Interactable {
 			GetComponentInChildren<MeshRenderer>().enabled = true;
 			Repaired?.Invoke();
 		}
+
+		public static Chair GetRandomEmptyChair() {
+			if (AllTables.Count == 0)
+				return null;
+
+			Table[] freeTables = AllTables.Where(tbl => !tbl.IsFull && !tbl.IsDestroyed).ToArray();
+			if (freeTables.Length == 0)
+				return null;
+
+			Table table = freeTables[Random.Range(0, freeTables.Length)];
+			Chair[] freeChairs = table.Chairs.Where(chr => !chr.IsOccupied).ToArray();
```

Comment:
Okay, so I've checked this & tested it and it works without any problems. However, I am curious and would like to understand how the program goes through the code here.
So, when the viking has randomly selected a table at line 151, if that table has no free chairs, does it loop around to check for another table then?
At first I read the code as if the viking would then leave, but they don't. They do find a free chair provided that there is one.

Reply:
The table selected at 151 does always have a chair free. This is ensured on 147 above. Not sure what to answer at the last paragraph.